tests/test_wemd.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import warnings
+from unittest import TestCase
+
+import numpy as np
+from numpy import all, asarray, ceil, cos, log2, mgrid, sin
+
+from aspire.operators import wemd_norm
+
+logger = logging.getLogger(__name__)
+
+
+def _smoothed_disk_image(x, y, radius, width, height):
+    (Y, X) = mgrid[:height, :width]
+    ratio = ((X - x) ** 2 + (Y - y) ** 2) / (radius ** 2)
+    return 2.0 - 2 / (1 + np.exp(-ratio))  # Sigmoid funciton
+
+
+def _is_monotone(seq):
+    arr = asarray(seq)
+    assert arr.ndim == 1
+    return all(arr[1:] >= arr[:-1])
+
+
+class WEMDTestCase(TestCase):
+    """
+    Test that the WEMD distance between smoothed disks of various radii,
+    angles and distances is monotone in the Euclidean distance of their centers.
+    Note that this monotonicity isn't strictly required by the theory,
+    but holds empirically.
+    """
+
+    def test_wemd_norm(self):
+        WIDTH = 64
+        HEIGHT = 64
+        CENTER_X = WIDTH // 2
+        CENTER_Y = HEIGHT // 2
+        # A rule of thumb for the wavelet level is to take
+        # the log2 of the side length.
+        LEVEL = int(ceil(log2(max(WIDTH, HEIGHT)))) + 1
+        WAVELET = "coif3"
+        warnings.filterwarnings(
```

Comment:
Why does the above rule of thumb generate warnings in `pywt`? Seems like this could be a problem.

Reply:
It's a truncation warning from PyWavelets, I don't think it indicates a real issue. If you want to not get this warning then you need to set a very low level, which gives bad results.

Reply:
In that case, why not filter out the warning in `wemd_embed`?

Reply:
I guess you're right. I've moved it.